tests/Model/FieldModelTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Opstalent\CrudBundle\Tests\Annotation;
+
+use Opstalent\CrudBundle\Exception\NotAllowedTypeException;
+use Opstalent\CrudBundle\Model\Field;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+/**
+ * Class FieldModelTest
+ * @author Szymon Kunowski <szymon.kunowski@gmail.com>
+ * @package Opstalent\CrudBundle
+ */
+class FieldModelTest extends TestCase
+{
+    /**
+     * @group FieldModel
+     * @group FormFactory
+     * @test
+     */
+    public function setProperName()
+    {
+        $model = new Field();
+        $this->assertInstanceOf(Field::class, $model->setName('user'));
+        $this->assertEquals('user', $model->getName());
+    }
+
+    /**
+     * @group FieldModel
+     * @group FormFactory
+     * @test
+     */
+    public function exceptionOnNonStringName()
```

Comment:
`throwsTypeErrorWhenNameIsNotAString`

Reply:
I won't be writing it for all further methods, I think you can see what I mean by previous ones :)